torch/csrc/autograd/functions/accumulate_grad.cpp

```diff
@@ -83,20 +83,21 @@ variable_list AccumulateGrad::apply_with_saved(
   at::Tensor variable_copy = variable;
   at::Tensor grad_copy = variable.grad();
   saved.before(variable_copy);
+  at::Tensor fake_variable_copy = variable_copy;
   saved.before(grad_copy);
   variable_copy.mutable_grad() = grad_copy;
   // op is intentionally static
   static auto op = c10::Dispatcher::singleton()
                        .findSchemaOrThrow("inductor::accumulate_grad_", "")
                        .typed<void(const at::Tensor&, const at::Tensor&)>();
   op.call(variable_copy, grads[0]);
+  auto& hook = tensor_post_acc_grad_hooks();
+  if (hook != nullptr) {
+    (*hook)(variable_copy);
+  }
```

Comment:
This seems wrong.  It will just trace through the hook (which might not be tracable).

Need something like:
```
hook->apply_with_saved(variable_copy, saved)
```

Then inside the handler for apply_with_saved the hooks is lifted to an input.

Reply:
Asked offline, I do not understand.